ios/Classes/SwiftHomeWidgetPlugin.swift

```diff
@@ -30,7 +36,14 @@ public class SwiftHomeWidgetPlugin: NSObject, FlutterPlugin, FlutterStreamHandle
       name: "home_widget/updates", binaryMessenger: registrar.messenger())
     eventChannel.setStreamHandler(instance)
 
-    registrar.addApplicationDelegate(instance)
+    guard isRunningInAppExtension() == false else {
+      return
+    }
```

Comment:
I'm not a swift/iOS expert so could you maybe elaborate how this (and checking for the extension via the extension) is different to wrapping the function invocation with an `#available(OSApplicationExtension, unavailable) check?

Reply:
Unfortunately, `#available` cannot be used to detect the target type. The code added inside `#available(OSApplicationExtension unavailable)` will not run anywhere. Checking the path extension is the only way to detect that code is being executed from an extension, not the main app, at least as far as I know.

The only alternative I can think of would be to create a separate configuration method that executes `addApplicationDelegate` and require users to call it from `didFinishLaunchingWithOptions`. However, I think that this would be a worse solution to this problem.